src/app/rpmostree-builtin-kargs.c

```diff
@@ -0,0 +1,143 @@
+  { "import-proc-cmdline", 0, 0, G_OPTION_ARG_NONE, &opt_import_proc, "Instead of modifying old kernel arguments, we modify args from current /proc/cmdline", NULL },
+  { "editor", 0, 0, G_OPTION_ARG_NONE, &opt_editor, "Pops up an editor for users to change arguments", NULL },
+  { NULL }
+};
+
+static GVariant *
+get_kargs_option_variant (void)
+{
+  GVariantDict dict;
+
+  g_variant_dict_init (&dict, NULL);
+  g_variant_dict_insert (&dict, "import-proc-cmdline", "b", opt_import_proc);
+
+  return g_variant_dict_end (&dict);
+}
+
+int
+rpmostree_ex_builtin_kargs (int            argc,
+                            char         **argv,
+                            RpmOstreeCommandInvocation *invocation,
+                            GCancellable  *cancellable,
+                            GError       **error)
+{
+  _cleanup_peer_ GPid peer_pid = 0;
+  glnx_unref_object RPMOSTreeSysroot *sysroot_proxy = NULL;
+  g_autoptr(GOptionContext) context = g_option_context_new("");
+  if (!rpmostree_option_context_parse (context,
+                                       option_entries,
+                                       &argc, &argv,
+                                       invocation,
+                                       cancellable,
+                                       NULL, NULL,
+                                       &sysroot_proxy,
+                                       &peer_pid,
+                                       error))
+    return EXIT_FAILURE;
+
+  if (opt_editor && (opt_import_proc || opt_kernel_delete_strings ||
+      opt_kernel_replace_strings ||  opt_kernel_append_strings))
+    {
+      /* We want editor command to achieve all the other functionalities */
+      /* Thus erroring out ahead of time when multiple options exist */
+      g_set_error (error, G_IO_ERROR, G_IO_ERROR_INVALID_ARGUMENT,
+                   "Cannot specify --editor with other options");
+      return EXIT_FAILURE;
+    }
+  if (opt_kernel_delete_strings && opt_kernel_replace_strings)
+    {
+      g_set_error (error, G_IO_ERROR, G_IO_ERROR_INVALID_ARGUMENT,
+                   "Cannot specify both --delete and --replace");
+      return EXIT_FAILURE;
+    }
+  if (opt_kernel_delete_strings && opt_kernel_append_strings)
+    {
+      g_set_error (error, G_IO_ERROR, G_IO_ERROR_INVALID_ARGUMENT,
+                   "Cannot specify both --delete and --append");
+      return EXIT_FAILURE;
+    }
+
+  /* We do not want editor option to be handled via daemon */
+  if (opt_editor)
+    {
+      ;
+    }
+  else
+    {
+      glnx_unref_object RPMOSTreeOS *os_proxy = NULL;
+      if (!rpmostree_load_os_proxy (sysroot_proxy, opt_osname,
+                                    cancellable, &os_proxy, error))
+        return EXIT_FAILURE;
+
+      g_autofree char *transaction_address = NULL;
+      char *empty_strv[] = {NULL};
+
+      /* dbus does not allow empty string arrays, assign it
+       * to be empty string array here to prevent erroring out */
```

Comment:
"does not allow NULL to mean the empty string array" would be a bit more precise.